deals/module/deals.go

```diff
@@ -298,23 +314,21 @@ func (m *Module) retrieve(ctx context.Context, lapi *apistruct.FullNodeStruct, l
 	return o.Miner.String(), out, nil
 }
 
-// GetDealStatus returns the current status of the deal, and a flag indicating if the miner of the deal was slashed.
-// If the deal doesn't exist, *or has expired* it will return ErrDealNotFound. There's not actual way of distinguishing
-// both scenarios in Lotus.
-func (m *Module) GetDealStatus(ctx context.Context, pcid cid.Cid) (storagemarket.StorageDealStatus, bool, error) {
-	lapi, cls, err := m.clientBuilder()
+// GetDealStatus returns the current status of the deal.
+func (m *Module) GetDealStatus(ctx context.Context, pcid cid.Cid) (storagemarket.StorageDealStatus, error) {
+	lapi, cls, err := m.clientBuilder(ctx)
 	if err != nil {
-		return storagemarket.StorageDealUnknown, false, fmt.Errorf("creating lotus client: %s", err)
+		return storagemarket.StorageDealUnknown, fmt.Errorf("creating lotus client: %s", err)
 	}
 	defer cls()
-	di, err := lapi.ClientGetDealInfo(ctx, pcid)
+	di, err := robustClientGetDealInfo(ctx, lapi, pcid)
```

Comment:
Considering that `ClientGetDealInfo` can lie about the real status of the deal, I created `robustClientGetDealInfo` with a workaround logic. All places that used `ClientGetDealInfo` now use `robustClientGetDealInfo`.